File: scotts_old_java_examples/JetpackDataBindingV2/app/src/main/java/com/ebookfrenzy/jetpackdatabindingv2/ui/main/MainViewModel.java
package com.ebookfrenzy.jetpackdatabindingv2.ui.main;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainViewModel extends ViewModel {
    public MutableLiveData<String> outputValue = new MutableLiveData<>();

    public void onTextChanged(CharSequence s, int start, int before, int count) {
           outputValue.setValue(String.valueOf(s));
    }

}

File: scotts_kotlin_examples/TwoWayDataBindingButton/app/src/main/java/com/scottshaper/twowaydatabindingbutton/ui/main/MainViewModel.kt
package com.scottshaper.twowaydatabindingbutton.ui.main

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class MainViewModel : ViewModel() {
    var inputText: MutableLiveData<String> = MutableLiveData()
    var outputText: MutableLiveData<String> = MutableLiveData()

    fun changeText(){
        outputText.value = inputText.value
    }
}